ТЗ 13 на проверку. Основной функцианал.

Konstakox opened the pull request:
Сервис для шеринга вещей. Каркас приложения, а также часть его веб-слоя.
Создание DTO-объектов и мапперов. 
Разработка контроллеров.
X-Sharer-User-Id — это идентификатор пользователя.

Konstakox committed:
ТЗ 13 на проверку. Основной функцианал.

Konstakox committed:
ТЗ 13 на проверку. Основной функцианал.
Исправление наименования класса.
Error:  src/main/java/ru/practicum/shareit/user/UserServiceImpl.java:[15,34] (naming) MemberName: Member name 'UserRepository' must match pattern '^[a-z][a-zA-Z0-9]*$'.

chehovnp commented on src/main/java/ru/practicum/shareit/item/ItemController.java:
```diff
@@ -1,12 +1,50 @@
+import java.util.List;
 
 /**
  * TODO Sprint add-controllers.
```
фразочка уже не нужная!

Konstakox replied:
Да. Удалил.

chehovnp commented on src/main/java/ru/practicum/shareit/item/ItemController.java:
```diff
@@ -1,12 +1,50 @@
 @RestController
+@RequiredArgsConstructor
 @RequestMapping("/items")
 public class ItemController {
```
добавь логирование запросов и его параметров

Konstakox replied:
Выполнил.

chehovnp commented on src/main/java/ru/practicum/shareit/item/ItemController.java:
```diff
@@ -1,12 +1,50 @@
+    private final ItemService itemService;
+
+    @PostMapping
+    public ItemDto addItem(@RequestHeader("X-Sharer-User-Id") Integer userId,
```
"X-Sharer-User-Id"  перенеси в константу ,  можно в отдельный класс

Konstakox replied:
Вынес в константу в отдельный класс

chehovnp commented on src/main/java/ru/practicum/shareit/item/ItemMapper.java:
```diff
@@ -0,0 +1,31 @@
+        );
+    }
+
+    // не знаю как правильно, удобнее builder...
```
как удобнее так и делай. 
сама переменная Item item не нужна , можно сразу : 
return Item.builder()
                .name(itemDto.getName())
                .description(itemDto.getDescription())
                .available(itemDto.getAvailable())
                .owner(userId)
                .request(itemDto.getRequest() != null ? itemDto.getRequest() : null)
                .build();

Konstakox replied:
Изменил.

chehovnp commented on src/main/java/ru/practicum/shareit/user/UserController.java:
```diff
@@ -1,12 +1,46 @@
 @RestController
+@RequiredArgsConstructor
 @RequestMapping(path = "/users")
 public class UserController {
```
логирование

Konstakox replied:
Выполнил.

chehovnp commented on src/main/java/ru/practicum/shareit/user/UserController.java:
```diff
@@ -1,12 +1,46 @@
+import java.util.List;
 
 /**
  * TODO Sprint add-controllers.
```
везде такие бы повесить !
шутка)

Konstakox replied:
уже висели в шаблоне, сразу решил не удалять. Потом глаза не мозолили). Да и напоминалка, что в каком спринте сделано/нужно сделать.
Выполненное удалил.

chehovnp commented on src/main/java/ru/practicum/shareit/user/UserMapper.java:
```diff
@@ -0,0 +1,25 @@
+public class UserMapper {
+
+    public static UserDto toUserDto(User user) {
+        UserDto userDto = UserDto.builder()
```
UserDto userDto  не нужна нам , сразу return

Konstakox replied:
исправил.

chehovnp commented on src/main/java/ru/practicum/shareit/user/UserRepositoryImpl.java:
```diff
@@ -0,0 +1,61 @@
+
+    @Override
+    public Boolean existUserWithEmail(String emailDto) {
+        return users.values().stream().anyMatch(user -> user.getEmail().equals(emailDto));
```
на данном этапе лучше емейлы хранить в репозитории в Set<String>

Konstakox replied:
Да, согласен. Но, в следующем задании нужно хранить в базе. Хранение в памяти вроде не будет использоваться. Не исправил.

chehovnp replied:
ок)

chehovnp reviewed (commented):
Константин, отправляю комменты ) 
жду правки.  Как изменишь , то ставь resolved на коммент и желательно ответ или мнение с чем не согласен )

Konstakox committed:
ТЗ 13 на проверку 2. Удалил выполненые тодошки. Добавил логирование запросов ItemController и UserController. Вынес X-Sharer-User-Id в константу.

Konstakox committed:
checkstyle
Добавил пустую строку
Error:  src/main/java/ru/practicum/shareit/constantsShareit/Constants.java:[8,9] (whitespace) EmptyLineSeparator: 'VARIABLE_DEF' should be separated from previous line.
Добавил пробел
Error:  src/main/java/ru/practicum/shareit/item/ItemServiceImpl.java:[58,80] (whitespace) NoWhitespaceBefore: ',' is preceded with whitespace.

chehovnp commented:
Спасибо за обратную связь. Жду следующую работу!
